docs/pubsub-api.rst

```diff
@@ -0,0 +1,255 @@
+``gcloud.pubsub`` API
+=====================
+
+Connection / Authorization
+--------------------------
+
+- Inferred defaults used to create connection if none configured explicitly:
+
+  - credentials (derived from GAE / GCE environ if present).
+
+  - ``project_id`` (derived from GAE / GCE environ if present).
+
+  - ``scopes``
+
+
+Manage topics for a project
+---------------------------
+
+Create a new topic for the default project:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.create_topic("topic_name")
+   >>> topic.name
+   'topic_name'
+
+Create a new topic for an explicit project:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.create_topic("topic_name", project_id="my.project")
+   >>> topic.name
+   'topic_name'
+
+Fetch an extant topic for the default project:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> topic.name
+   'topic_name'
+
+Fetch an extant topic for the default project:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name', project_id="my.project")
+   >>> topic.name
+   'topic_name'
+
+Attempt to fetch a non-extant topic:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> pubsub.get_topic('nonesuch')
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in ?
+   NotFound: ...
+
+List extant topics for the default project:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> [topic.name for topic in pubsub.list_topics()]
+   ['topic_name']
+
+List extant topics for an explicit project:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> [topic.name for topic in pubsub.list_topics(project_id="my.project")]
+   ['topic_name']
+
+Delete a topic:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> topic.delete()
+
+
+Publish messages to a topic
+---------------------------
+
+Publish a single message to a topic, without attributes:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> topic.publish('this is the message_payload')
+   <message_id>
+
+Publish a single message to a topic, with attributes:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> topic.publish('this is another message_payload',
+   ...               attr1='value1', attr2='value2')
+   <message_id>
+
+Publish a set of messages to a topic (as a single request):
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> with topic:
+   ...     topic.publish('this is the first message_payload')
+   ...     topic.publish('this is the second message_payload',
+   ...                   attr1='value1', attr2='value2')
+   [<message_id1>, <message_id2>]
```

Comment:
OK [double checked the docs](https://cloud.google.com/pubsub/reference/rest/v1beta2/projects/topics/publish). We're good "The server-assigned ID of each published message, in the same order as the messages in the request. IDs are guaranteed to be unique within the topic."